### DISCUSSION TITLE
v5.65.0

### DISCUSSION THREAD

**Discussion by sokra** [webpack/webpack#14916] [General]
# Features

* static evaluation understands `undefined` now
* reduce container entry code by a few chars
* use template literals when available and they make sense

# Bugfixes

* handle `singleton` flag without `requiredVersion` in Module Federation
* upgrade `watchpack` for context time info bugfix

# Performance

* improve RegExp in error message formating for non-quadratic performance

# Developer Experience

* automatically insert brackets when `output.globalObject` contains a non-trival expression
* show error when using `script` type external with invalid syntax
* expose types for `Resolver`, `StatsOptions` and `ResolvePluginInstance`

# Preparations for the future

* `hashDigestLength` will default to 16 in webpack 6 (`experiments.futureDefaults`)

<hr /><em>This discussion was created from the release <a href='https://github.com/webpack/webpack/releases/tag/v5.65.0'>v5.65.0</a>.</em>

**Comment by POOHBEAR77l** [comment]
Awesome

**Comment by POOHBEAR77l** [comment]
> # Features
> 
> * static evaluation understands `undefined` now
> * reduce container entry code by a few chars
> * use template literals when available and they make sense
> 
> # Bugfixes
> 
> * handle `singleton` flag without `requiredVersion` in Module Federation
> * upgrade `watchpack` for context time info bugfix
> 
> # Performance
> 
> * improve RegExp in error message formating for non-quadratic performance
> 
> # Developer Experience
> 
> * automatically insert brackets when `output.globalObject` contains a non-trival expression
> * show error when using `script` type external with invalid syntax
> * expose types for `Resolver`, `StatsOptions` and `ResolvePluginInstance`
> 
> # Preparations for the future
> 
> * `hashDigestLength` will default to 16 in webpack 6 (`experiments.futureDefaults`)
> 
> 
> <hr /><em>This discussion was created from the release <a href='https://github.com/webpack/webpack/releases/tag/v5.65.0'>v5.65.0</a>.</em>
